src/metabase/db/setup.clj

```diff
@@ -52,7 +56,29 @@
                    "\n"))
     (throw (Exception. (trs "Database requires manual upgrade.")))))
 
-(s/defn migrate!
+(mu/defn ^:private initialize-db!
```

Comment:
one problem with this is we might run into race condition if someone deploy metabase as multi-instances.

liquibase itself used lock to avoid that but we're running outside of liqubiase context here.

Should we worry about that?